enumerable_methods.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i], i)
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_select
+    ary = []
+    if block_given?
+      my_each { |elem| ary.push(elem) if yield(elem) }
+    else
+      'No block given'
+    end
+    ary
+  end
+
+  def my_all?
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+    else
+      my_each { |elem| return false unless elem }
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      my_each { |elem| return true if yield(elem) }
+    else
+      my_each { |elem| return true if elem }
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      my_each { |elem| return false if yield(elem) }
+    else
+      my_each { |elem| return false if elem }
+    end
+    true
+  end
+
+  def my_count
+    c = 0
+    if block_given?
+      my_each { |elem| c += 1 if yield(elem) }
+    else
+      my_each { c += 1 }
+    end
+    c
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] counts the number of items in enum that are equal to item if an argument is given